### PR TITLE
fix(files): switched native input to NcButton

### LINKED ISSUE #42593: [BITV]: Replace "Transfer" button with NcVue button
Contrast issue, html valid elements issue.

Replace `<input data-v-67866491="" type="submit" disabled="disabled" class="primary" value="Transfer">` with NcVue button.

Screenshot from 2024-01-05 11-21-52

### PR DESCRIPTION
**Pull request by emoral435**
* Resolves: #42593
## Summary
Replaces the 'input' button to transfer files / ownership with our native NcButton component

## New DOM Update View
image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by emoral435** [comment]
/backport 86a8e078bd362392cda5a153120f2318e567aa4a to stable28

File: apps/files/src/components/TransferOwnershipDialogue.vue [hunk 1]
```diff
@@ -50,10 +50,13 @@
 					@search="findUserDebounced" />
 			</p>
 			<p>
-				<input type="submit"
-					class="primary"
-					:value="submitButtonText"
+				<NcButton nativeType="submit"
```
Comment (Pytal) [inline apps/files/src/components/TransferOwnershipDialogue.vue:53]:
```suggestion
				<NcButton native-type="submit"
```
Kebab-case for props ;)

File: apps/files/src/components/TransferOwnershipDialogue.vue [hunk 2]
```diff
@@ -50,10 +50,13 @@
...
+					type="primary"
 					:disabled="!canSubmit">
+					<template #default>
+						{{ submitButtonText }}
+					</template>
```
Comment (Pytal) [inline apps/files/src/components/TransferOwnershipDialogue.vue:58]:
```suggestion
					{{ submitButtonText }}
```
Don't need to pass default template explicitly

**Review by ShGKme — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

**Pull request merged**